### ISSUE TITLE
Pretty permalink structure

### ISSUE THREAD

**Issue by simonszu** [bitnami/bitnami-docker-dokuwiki#39]
**Description**

I'd like to activate pretty permalinks. The method .htaccess results in dokuwiki not being able to parse the permalinks. 

**Steps to reproduce the issue:**

1. Open Dokuwiki configuration
2. Set url rewriting to ".htaccess"
3. Refresh page several times

**Describe the results you received:**

`The requested url /start cannot be found`

**Describe the results you expected:**

Pretty permalinks which are resolved by the container's apache 

**Additional information you deem important (e.g. issue happens only occasionally):**

I am proxying with an nginx to the container's apache. 

**Version**

- Output of `docker version`:

```
Docker version 17.12.1-ce, build 7390fc6

```

- Output of `docker info`:

```
Containers: 3
 Running: 3
 Paused: 0
 Stopped: 0
Images: 11
Server Version: 17.12.1-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 9b55aab90508bd389d7654c4baf173a981477d55
runc version: 9f9c96235cc97674e935002fc3d78361b696a69e
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.0-6-amd64
Operating System: Debian GNU/Linux 9 (stretch)
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 5.832GiB
Name: tharan
ID: LIOZ:XB3C:EECZ:DJGA:6GZD:TSDG:3GSP:XSOE:7BVJ:QC7W:ZRZX:7LEY
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Username: simonszu
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support

```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**

Debian host system, no swap, nginx proxy to the containers, nginx handles SSL stuff and proxies to container's port 80.

**Comment by juan131** [comment]
Hi @simonszu 

All the htaccess files of Dokuwiki are collected on a single file called **/opt/bitnami/apache/conf/vhosts/htaccess/dokuwiki-htaccess.conf** and we're setting the directive **AllowOverride** to **"none"** at the configuration file **/opt/bitnami/apache/conf/vhosts/dokuwiki-vhost.conf**.

> NOTE: The Apache Software Foundation also recommends [this configuration](http://httpd.apache.org/docs/current/mod/core.html#allowoverride). "For security and performance reasons, do not set AllowOverride to anything other than None in your block. Instead, find (or create) the block that refers to the directory where you're actually planning to place a .htaccess file."

In order to user **.htaccess** files, you need to edit the configuration file **/opt/bitnami/apache/conf/vhosts/dokuwiki-vhost.conf** so you change the **AllowOverride** directive.

**Comment by simonszu** [comment]
Thanks. I have enabled `AllowOverride` now, so that i get prettier URLs.

**Comment by mauvehed** [comment]
I'm just working through this same thing, seems like there needs to be some method to allow rewrites even if it's not allowed through a local .htacess file.

**Comment by juan131** [comment]
Hi @mauvehed 

*AllowOverride* is disabled on the Apache configuration as you can check on the configuration file **/opt/bitnami/apache/conf/vhosts/dokuwiki-vhost.conf**:

```
$ docker-compose exec dokuwiki cat /opt/bitnami/apache/conf/vhosts/dokuwiki-vhost.conf
...
  <Directory "/opt/bitnami/dokuwiki">
    Options -Indexes +FollowSymLinks -MultiViews
    AllowOverride None
    Require all granted
    DirectoryIndex index.html index.php


  </Directory>
...
```

**Comment by svenssonaxel** [comment]
See my comment on #49

**Comment by juan131** [comment]
I just added some comments @svenssonaxel

**Comment by marcosbc** [comment]
Hi all, just wanted to let you know that we have just released DokuWiki `20200729.0.0-debian-10-r6` which enables nice/friendly URLs by default.
